src/main/service/listManipulation/OpaPublicListService.ts

```diff
@@ -5,22 +5,25 @@ import { OpaPressListService } from './OpaPressListService';
 const opaPressListService = new OpaPressListService();
 
 export class OpaPublicListService {
-    public formatOpaPublicList(opaPublicListJson: string): any {
+    public formatOpaPublicList(jsonData: string): any {
         const rows = [];
-        JSON.parse(opaPublicListJson).courtLists.forEach(courtList => {
+
+        JSON.parse(jsonData).courtLists.forEach(courtList => {
             courtList.courtHouse.courtRoom.forEach(courtRoom => {
                 courtRoom.session.forEach(session => {
                     session.sittings.forEach(sitting => {
                         sitting.hearing.forEach(hearing => {
-                            const allDefendants = this.processPartyRoles(hearing);
-                            allDefendants.forEach(defendant => {
-                                hearing.case.forEach(hearingCase => {
-                                    const caseDetails = this.buildCaseDetails(hearingCase);
-                                    rows.push({
-                                        ...caseDetails,
-                                        ...defendant,
+                            hearing.case.forEach(hearingCase => {
+                                if (hearingCase['party']) {
```

Comment:
Do not need the if statement here. Within the method processPartyRoles(), it is using "hearingCase.party?.forEach(...)". The question mark after the party is already checking whether party exist or not

Reply:
Removed if statement. Thanks Kian